concrete/src/Cache/Page/PageCache.php

```diff
@@ -191,33 +191,29 @@ public function shouldAddToCache(PageView $v)
     /**
      * Get the key that identifies the cache entry for a page or a request.
      *
-     * @param \Concrete\Core\Page\Page|\Concrete\Core\Http\Request|mixed $mixed
+     * @param \Concrete\Core\Page\Page|\Concrete\Core\Http\Request|\Concrete\Core\Cache\Page\PageCacheRecord|mixed $mixed
      *
      * @return string|null Returns NULL if $mixed is not a recognized type, a string otherwise
      */
     public function getCacheKey($mixed)
     {
-        $homeCID = 1;
         if ($mixed instanceof ConcretePage) {
-            if ($mixed->getCollectionPath() != '') {
-                return urlencode(trim($mixed->getCollectionPath(), '/'));
-            } else {
-                if ($mixed->getCollectionID() == $homeCID) {
-                    return '!' . $homeCID;
-                }
+            $collectionPath = trim((string) $mixed->getCollectionPath(), '/');
+            if ($collectionPath !== '') {
+                return urlencode($collectionPath);
+            }
+            $cID = $mixed->getCollectionID();
+            if ($cID  && $cID == ConcretePage::getHomePageID()) {
+                return '!' . $cID;
             }
-        } else {
-            if ($mixed instanceof \Concrete\Core\Http\Request) {
-                if ($mixed->getPath() != '') {
-                    return urlencode(trim($mixed->getPath(), '/'));
-                } else {
-                    return '!' . $homeCID;
-                }
-            } else {
-                if ($mixed instanceof PageCacheRecord) {
-                    return $mixed->getCacheRecordKey();
-                }
+        } elseif ($mixed instanceof \Concrete\Core\Http\Request) {
```

Comment:
We shouldn't use the FQN here, this class should be `use`d

Reply:
Updated.